src/Containers/Microsoft.NET.Build.Containers/Tasks/ParseContainerProperties.cs

```diff
@@ -141,22 +141,19 @@ public override bool Execute()
             return !Log.HasLoggedErrors;
         }
 
-        try
+        var (normalizedImageName, normalizationWarning, normalizationError) = ContainerHelpers.NormalizeImageName(ContainerImageName);
+        if (normalizedImageName is not null)
         {
-            if (!ContainerHelpers.NormalizeImageName(ContainerImageName, out var normalizedImageName))
-            {
-                Log.LogMessageFromResources(nameof(Strings.NormalizedContainerName), nameof(ContainerImageName), normalizedImageName);
-                NewContainerImageName = normalizedImageName!; // known to be not null due to output of NormalizeImageName
-            }
-            else
-            {
-                // name was valid already
-                NewContainerImageName = ContainerImageName;
-            }
+            NewContainerImageName = normalizedImageName;
+        }
+        if (normalizationWarning is (string warningMessageKey, object[] warningParams))
+        {
+            Log.LogMessageFromResources(warningMessageKey, warningParams);
         }
-        catch (ArgumentException)
+
+        if (normalizationError is (string errorMessageKey, object[] errorParams))
         {
-            Log.LogErrorWithCodeFromResources(nameof(Strings.InvalidContainerImageName), nameof(ContainerImageName), ContainerImageName);
+            Log.LogErrorWithCodeFromResources(errorMessageKey, errorParams);
```

Comment:
Instead of incorrectly forwarding the error message to the user via the ArgumentException, we now just let the Normalize call tell us what to do with the messages directly.

Warnings are allowed to continue execution, while errors might end up stopping the execution.